nats.go

```diff
@@ -3790,6 +3790,89 @@ func (m *Msg) RespondMsg(msg *Msg) error {
 	return nc.PublishMsg(msg)
 }
 
+func (m *Msg) isJSMsg() bool {
+	if m.Reply == "" {
+		return false
+	}
+
+	parts := strings.Split(m.Reply, ".")
+	if len(parts) == 8 && parts[0] == "$JS" && parts[1] != "ACK" {
+		return true
+	}
+
+	return false
+}
+
+var (
+	AckAck      = []byte("+ACK")
+	AckNak      = []byte("-NAK")
+	AckProgress = []byte("+WPI")
+	AckNext     = []byte("+NXT")
+	AckTerm     = []byte("+TERM")
+)
+
+// Ack acknowledges a JetStream messages received from a Consumer, indicating the message
+// should not be received again later
+func (m *Msg) Ack() error {
+	if !m.isJSMsg() {
```

Comment:
Not sure we care, if app thinks its acking we should just send reply unless there is no reply subject. Ack subject may change etc and this could be brittle.